mlflow/models/model_config.py

```diff
@@ -13,17 +13,17 @@ class ModelConfig:
     overridden when logging a model.
     """
 
-    def __init__(self, *, development_config: Optional[str] = None):
+    def __init__(self, *, development_config: Optional[Union[str, Dict[str, Any]]] = None):
         config = globals().get("__mlflow_model_config__", None)
         # backwards compatibility
         rag_config = globals().get("__databricks_rag_config_path__", None)
-        self.config_path = config or rag_config or development_config
+        self.config = config or rag_config or development_config
 
-        if not self.config_path:
+        if not self.config:
             raise FileNotFoundError("Config file is None. Please provide a valid path.")
 
-        if not os.path.isfile(self.config_path):
-            raise FileNotFoundError(f"Config file '{self.config_path}' not found.")
+        if not isinstance(development_config, dict) and not os.path.isfile(self.config):
+            raise FileNotFoundError(f"Config file '{self.config}' not found.")
```

Comment:
Can we replace this `if not isinstance(development_config, dict)` with `if not isinstance(self.config, dict)`
Since `self.config` should take precedence and we should check if its a file or not even if the `development_config, dict` is a dict.
This is useful when loading the model back.

Reply:
Yes, agreed should be self.config. Will update!